syntax = "proto3";

option csharp_namespace = "GrpcCloud";


enum Status {
    SUCCESS = 0;
    FAILURE = 1;
}

message LeaderToViewerHeartBeatRequest
{
	int32 term = 1;
	int32 systemLastIndex = 2;
	string LeaderIP = 3; //or leader id
}

message LeaderToViewerHeartBeatResponse
{
	bool status = 1;
	optional string message = 2;
}

service Cloud {

	//Raft:
	rpc GetOrUpdateSystemLeader(LeaderToViewerHeartBeatRequest) returns (LeaderToViewerHeartBeatResponse);
}
